modules/core/shared/src/main/scala/io/circe/Error.scala

```diff
@@ -52,32 +52,36 @@ sealed abstract class DecodingFailure(val reason: Reason) extends Error {
 
   def history: List[CursorOp]
 
-  val message: String = reason match {
-    case WrongTypeExpectation(expJsType, v) => s"Got value '${v.noSpaces}' with wrong type, expecting $expJsType"
-    case MissingField                       => "Missing required field"
-    case CustomReason(message)              => message
-  }
+  /**
+   * The location in the JSON where the decoding failure occurred. For various
+   * reasons, in 0.14.x this may not always be present.
+   */
+  def pathToRootString: Option[String] = None
+
+  final def message: String =
```

Comment:
I think MiMa is complaining about this. Since this is a `sealed` class I think this would be okay to add an exception for. But it's also easy enough to remove the `final` modifier.

```suggestion
  def message: String =
```

Reply:
I think I'd prefer removing `final` rather than adding an exception.

Reply:
Updated